### PR TITLE
Port int8 and int16 in ABS kernel to micro

### LINKED ISSUE #190: Port operator ABS for int8 and int16 to TFLu
@advaitjain This tracks porting of int8 and int16 support in ABS kernel.

It will be delivered as one PR.

### PR DESCRIPTION
**Pull request by patriklaurell**
@advaitjain this PR ports int8 and int16 support to ABS kernel.

BUG=Fixes #190

### CODE REVIEW

**Comment by patriklaurell** [comment]
@advaitjain all tests passed. Ready for review

File: tensorflow/lite/micro/kernels/elementwise.cc [hunk 1]
```diff
@@ -27,6 +29,23 @@ namespace micro {
...
+constexpr char kCosName[] = "Cos";
+constexpr char kLogName[] = "Log";
+constexpr char kSqrtName[] = "Sqrt";
+constexpr char kRsqrtName[] = "Rsqrt";
+constexpr char kSquareName[] = "Square";
+constexpr char kNotName[] = "Not";
+
+struct OpData {
+  int32_t multiplier;
+  int shift;
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/elementwise.cc:43]: nit: int32_t

File: tensorflow/lite/micro/kernels/elementwise.cc [hunk 2]
```diff
@@ -35,8 +54,19 @@ bool IsLogicalSupportedType(const TfLiteType type) {
   return type == kTfLiteBool;
 }
 
+bool IsAbsSupportedType(const TfLiteType type) {
+  return type == kTfLiteFloat32 || type == kTfLiteInt8 || type == kTfLiteInt16;
+}
+
+inline void SetAbsOutputMultiplier(const float input_scale,
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/elementwise.cc:61]: nit: remove inline

**Review by advaitjain — CHANGES_REQUESTED** [review]
There is a lot of somewhat subtle code that can be shared between the TFL and TFLM implementation. It appears that other than the Init functions almost everything else can be shared?

I think we should refactor the TFL kernel to move the shared code into tensorflow/lite/kernels/elementwise_portable.cc and elementwise_portable.h

and then use the shared code from micro/kernels/elementwise.cc

The TFL refactoring will have to go in the tensorflow repo.

**Comment by patriklaurell** [comment]
@advaitjain That sounds like a good idea. 

So to clarify. I move all the shared code into elementwise_portable.cc and elementwise_portable.h, and submit that as a PR to the tensorflow repo. Once that PR is merged will it automatically be merged into this repo or will I need to create an additional PR to merge the elementwise_portable change into this repo?

**Comment by advaitjain** [comment]
> @advaitjain That sounds like a good idea.
> 
> So to clarify. I move all the shared code into elementwise_portable.cc and elementwise_portable.h, and submit that as a PR to the tensorflow repo. Once that PR is merged will it automatically be merged into this repo or will I need to create an additional PR to merge the elementwise_portable change into this repo?

Apologies for the delay.

Yes, let's make a PR in the tensorflow repo for the shared code. Once it is merged, explicitly adding the shared code to https://github.com/tensorflow/tflite-micro/blob/a7f2a887bd947c7378d8eada2d0b82dd5a8c1ca9/ci/sync_from_upstream_tf.sh#L34-L36 will allow it to be sync'd to this repo -- I can help with the sync.

**Comment by patriklaurell** [comment]
@advaitjain I opened [this PR](https://github.com/tensorflow/tensorflow/pull/50836) for the refactoring of the elementwise kernel in TFLite.

I couldn't refactor out the eval code into elementwise_portable.h because Lite uses std::function, whereas the micro code is written in a way to avoid that. If I am not mistaken this is done to avoid dynamic allocation. Should we keep it like this or should I change the lite code to be aligned with the way it is done in micro?

**Comment by patriklaurell** [comment]
After talking with @advaitjain we decided that the std::function uses in the Lite kernel should be removed and implemented as in Micro, so that more code can be shared between Lite and Micro. I am working on that refactoring now and will push it to [the PR in the main tensorflow repo ](https://github.com/tensorflow/tensorflow/pull/50836)

**Comment by advaitjain** [comment]
Closing old PR. Let's reopen if needed.

**Pull request closed without merging**